src/pydata_sphinx_theme/toctree.py

```diff
@@ -137,18 +137,22 @@ def generate_header_nav_html(n_links_before_dropdown: int = 5) -> str:
         out = "\n".join(links_solo)
 
         # Wrap the final few header items in a "more" dropdown
-        links_dropdown = links_html[n_links_before_dropdown:]
+        links_dropdown = [
+            html.replace("nav-link", "nav-link dropdown-item")
+            for html in links_html[n_links_before_dropdown:]
+        ]
+
         if links_dropdown:
             links_dropdown_html = "\n".join(links_dropdown)
             out += f"""
-            <div class="nav-item dropdown">
-                <button class="btn dropdown-toggle nav-item" type="button" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
+            <li class="nav-item dropdown">
+                <button class="btn dropdown-toggle nav-item" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-controls="pst-header-nav-more-links">
                     More
                 </button>
-                <div class="dropdown-menu">
+                <ul id="pst-header-nav-more-links" class="dropdown-menu">
```

Comment:
FWIW the version switcher button has ID `versionswitcherbutton` 🤷🏻  I sorta like that actually because mashing all the words together makes it obviously different from how we do class names.

Reply:
Hmm, that triggers the spidey sense from my third-party JavaScript background. :)

Without prefixes for names that live in a global namespace, it's so, so easy to end up with name clashes. What if someone uses this theme but one of their pages loads a third-party JavaScript widget that has its own version switcher button with the same id?

Reply:
> Without prefixes for names that live in a global namespace, it's so, so easy to end up with name clashes.

good point, please open a new issue that we should prefix all our `id` attributes with `pst_` or similar.

Reply:
https://github.com/pydata/pydata-sphinx-theme/issues/1417